Fix Travis CI tests with low deps

jaylinski opened the pull request:
Until now, the low deps test was useless because
we always executed a `composer update` afterwards,
which updated all installed low-deps packages to
its latest version.

Relates to #21

jaylinski commented on .travis.yml:
```diff
@@ -1,6 +1,6 @@
 language: php
 
-matrix:
```
The `matrix` keyword is deprecated: https://docs.travis-ci.com/user/build-matrix/

jaylinski commented on .travis.yml:
```diff
@@ -11,29 +11,27 @@ matrix:
-    - if [[ $deps = low ]]; then composer update --prefer-lowest --prefer-stable; fi
+install:
+  - travis_retry composer install --no-interaction
+  - if [[ $deps = low ]]; then travis_retry composer update --prefer-lowest --prefer-stable --no-interaction; fi
```
Here's how Laravel does it: https://github.com/laravel/framework/blob/v6.10.0/.travis.yml

shulard replied:
Thank you for the tip !

jaylinski commented on .scrutinizer.yml:
```diff
@@ -1,18 +1,18 @@
+  php_loc: true
+  php_pdepend: true
+  php_sim: true
+  php_changetracking: true
```
Formatting according to our `.editorconfig`.

jaylinski commented on composer.json:
```diff
@@ -39,8 +39,8 @@
-        "phpspec/phpspec": "^4.2 || ^5.0 || ^6.0",
-        "phpunit/php-code-coverage": "^5.0 || ^6.0 || ^7.0"
+        "phpspec/phpspec": "^5.0 || ^6.0",
+        "phpunit/php-code-coverage": "^6.0 || ^7.0"
```
This fixes the failing low-deps tests.

jaylinski commented on phpspec.yml:
```diff
@@ -2,9 +2,9 @@
-        psr4_prefix: FriendsOfPhpSpec\PhpSpec\CodeCoverage
+  default_suite:
+    namespace: FriendsOfPhpSpec\PhpSpec\CodeCoverage
+    psr4_prefix: FriendsOfPhpSpec\PhpSpec\CodeCoverage
```
Formatting according to our `.editorconfig`.

jaylinski commented on phpspec.yml:
```diff
@@ -2,18 +2,15 @@
-      html: build/coverage
-      clover: build/coverage.xml
-      php: build/coverage.php
+      html: coverage
```
I had to change this because of the `deps=low` test and this bug: https://github.com/sebastianbergmann/php-code-coverage/issues/584

shulard replied:
Maybe we can add the `phpunit/php-code-coverage` v6.0.0 in the conflict section of the composer.json ? This way we will disallow to install the failing version.

Also, why removing the PHP format ? I'm not sure why there is all these formats generated ^^

jaylinski replied:
I also thought about adding the package in the conflict section. But I think using a `coverage` folder instead of a `build` folder is a better naming (because we don't really build anything when running our tests).

I removed the PHP format because I couldn't find a place were we are using it.

shulard replied:
Ok you are right using the conflict section just because of a `dist` folder seems a bit useless :smile:.

jaylinski commented on composer.json:
```diff
@@ -39,8 +39,11 @@
+        "phpunit/php-code-coverage": "^6.0 || ^7.0"
+    },
+    "conflict": {
+        "sebastian/comparator": "<2.0"
```
I had to change this because of failing low-deps tests caused by this bug on PHP >=7.2: https://github.com/sebastianbergmann/comparator/pull/30

jaylinski replied:
@drupol This may also help you with #18.
```
"conflict": {
    "phpunit/php-code-coverage": "<6.0"
}
```

drupol replied:
Oooooh !!! Trying it out :-) Thanks mate!

jaylinski commented on .travis.yml:
```diff
@@ -11,29 +11,29 @@ matrix:
     - php: nightly
   allow_failures:
+    - php: 7.4
+      env: deps=low # This will install phpspec v5, which does not support PHP 7.4.
```
Maybe we should completely remove the `deps=low` test for PHP 7.4? :thinking:

shulard replied:
Are you sure it will install PHPSpec v5 since composer will check the PHP version during dependencies resolution ?

jaylinski replied:
You are right. This was another issue I addressed here: https://github.com/friends-of-phpspec/phpspec-code-coverage/pull/22#discussion_r365200439

shulard commented:
Thank you @jaylinski for this nice MR ! There is a lot tips to be saved for later projects ^^

Maybe you can make multiple commits to split the different changes ? For example the CS/Sniff fixes can be splitted from the Travis ones ?

drupol commented:
Definitely nice :-)

We should also evaluate the migration from Travis to Github Actions ([see my default configuration file](https://github.com/loophp/collection/blob/master/.github/workflows/continuous-integration.yml)).

jaylinski commented on composer.json:
```diff
@@ -39,8 +39,12 @@
+        "phpunit/php-code-coverage": "^6.0 || ^7.0"
+    },
+    "conflict": {
+        "friendsofphp/php-cs-fixer": "2.16.0",
```
I had to add this because of failing low-deps tests related to this commit on PHP 7.4:
https://github.com/FriendsOfPHP/PHP-CS-Fixer/commit/121e0a6135e654466fcd1163a78b423c4fffee55

jaylinski commented:
@shulard Thanks for your feedback! I split the PR in multiple commits and fixed the issue you mentioned in https://github.com/friends-of-phpspec/phpspec-code-coverage/pull/22#discussion_r364621659.

shulard commented:
Awesome, thank you @jaylinski !
